pkg/controller/v1alpha1/trainedmodel/controller.go

```diff
@@ -118,15 +122,79 @@ func (r *TrainedModelReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error
 		return ctrl.Result{}, nil
 	}
 
+	// update URL and Address fo TrainedModel
+	if err := r.updateStatus(req, tm); err != nil {
+		return ctrl.Result{}, err
+	}
+
 	// Reconcile modelconfig to add this TrainedModel to its parent InferenceService's configmap
 	if err := r.ModelConfigReconciler.Reconcile(req, tm); err != nil {
 		return ctrl.Result{}, err
 	}
 	return ctrl.Result{}, nil
 }
 
-func (r *TrainedModelReconciler) updateStatus(desiredService *v1alpha1api.TrainedModel) error {
-	//TODO update TrainedModel status object, this will be done in a separate PR
+func (r *TrainedModelReconciler) updateStatus(req ctrl.Request, desiredModel *v1alpha1api.TrainedModel) error {
+	// Get the parent inference service
+	isvc := &v1beta1api.InferenceService{}
+	if err := r.Get(context.TODO(), types.NamespacedName{Namespace: req.Namespace, Name: desiredModel.Spec.InferenceService}, isvc); err != nil {
+		return err
+	}
+
+	// obtain protocol version of predictor spec
+	protocolVersion := string(isvc.Spec.Predictor.GetImplementation().GetProtocol())
+
+	// V1 ends with :predict whereas V2 ends with /infer
+	endpointSuffix := ":predict"
+	if protocolVersion == string(constants.ProtocolV2) {
+		endpointSuffix = "/infer"
+	}
+
+	// Check if parent inference service has the status URL
+	if isvc.Status.URL != nil {
+		// Update status to contain the isvc URL with /v1/models/trained-model-name:predict appened
+		url := isvc.Status.URL.String() + "/" + protocolVersion + "/models/" + desiredModel.Name + endpointSuffix
```

Comment:
We can use [this utility](https://github.com/kubeflow/kfserving/blob/master/pkg/constants/constants.go#L282) to construct the predict path.